(2023.03.20) Peter Trekels (ptrekels@androme.com)
ONEM-29883 Fix asserts in wpeframework-ocdm-playready-nexus-svp callback

Index: OCDM-Playready-Nexus-SVP/MediaSessionExt.cpp
===================================================================
--- OCDM-Playready-Nexus-SVP.orig/MediaSessionExt.cpp
+++ OCDM-Playready-Nexus-SVP/MediaSessionExt.cpp
@@ -137,10 +137,9 @@ DRM_RESULT opencdm_output_levels_callback(
     const DRM_PLAY_OPL_EX2 * const opl = static_cast<const DRM_PLAY_OPL_EX2 *>(outputLevels);
 #else
     const DRM_PLAY_OPL_LATEST * const opl = static_cast<const DRM_PLAY_OPL_LATEST *>(outputLevels);
+    ASSERT(opl->dwVersion == VER_DRM_PLAY_OPL_LATEST);
 #endif
 
-    ASSERT(opl->dwVersion == 0);
-
     decryptContext->outputProtection.setOutputLevels(opl->minOPL);
 
     // MaxRes Decode
@@ -154,11 +153,14 @@ DRM_RESULT opencdm_output_levels_callback(
 #ifdef URSR_VERSION_LESS_OR_EQUAL_20
         const DRM_OUTPUT_PROTECTION_EX &dope = dvopi.rgVop[i];
 #else
+        ASSERT(dvopi.dwVersion == VER_DRM_DIGITAL_VIDEO_OUTPUT_PROTECTION_IDS_LATEST);
         const DRM_ANALOG_VIDEO_OUTPUT_PROTECTION_LATEST &dope = dvopi.rgVop[i];
 #endif
         if (DRM_IDENTICAL_GUIDS(&dope.guidId, &g_guidMaxResDecode))
         {
-            ASSERT(dope.dwVersion == 3);
+#ifndef URSR_VERSION_LESS_OR_EQUAL_20
+            ASSERT(dope.dwVersion == VER_DRM_DIGITAL_VIDEO_OUTPUT_PROTECTION_LATEST);
+#endif
             uint32_t mrdWidth, mrdHeight;
             const int inc = sizeof(uint32_t);
             ASSERT(dope.cbConfigData >= 2*inc);
